presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPartitionManager.java

```diff
@@ -78,6 +91,87 @@ public List<String> getEffectivePartitions(
                 partitionPredicate.put(partitionColumn, Domain.all(column.getHiveType().getType(typeManager)));
             }
         }
-        return metastore.getPartitionNamesByFilter(metastoreContext, schemaName, tableName, partitionPredicate);
+
+        // fetch partition names from metastore
+        List<String> partitionNames = metastore.getPartitionNamesByFilter(metastoreContext, schemaName, tableName, partitionPredicate);
+        List<Type> partitionTypes = partitionColumns.stream()
+                .map(column -> column.getType().getType(typeManager))
+                .collect(toList());
+        Constraint<ColumnHandle> constraint = new Constraint<>(tupleDomain);
+        // match partition and get partition only we need
+        Iterable<HudiPartition> partitionsIterable = () -> partitionNames.stream()
+                // Apply extra filters which could not be done by getFilteredPartitionNames
```

Comment:
getFilteredPartitionNames is not getting used in presto-hudi module but in presto-hive module. Just trying to understand if we can reuse the existing methods there, since the code looks more or less similar to me.

Reply:
Ah I see, we want to make use of HudiPartition here, hence the new methods. Let us either remove this comment or change it to `// Apply extra filters`? @harris233

Reply:
OK, I have removed it.